### ISSUE TITLE
Bug: [no-unnecessary-type-arguments] False negative on JSX

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#10629]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=GYVwdgxgLglg9mABAIRFKCA8AVRBeRAZygCcYwBzAPgAoBKRAbwChFESBTKEEpTAExgA3KpgD0gkQG5mAX2YQExRACM0GJAUyp0WYmUpVEYqlONiiACzggANv3YcADnBJRENYAENbhDojAOCi9YIQ46BSV3NV0wACZ8FHU9UnJqejMxCw4ADycOaA5%2BZiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWiIDmsALaIWqDJETRoHaJHBgAviF1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```tsx
function Button<T = string>() {
  return <div></div>;
}
const button = <Button<string> />; // should report (false negative)
const button2 = Button<string>(); // expected
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-type-arguments": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

line 4, ` <Button<string> />;` should be reported.

### Actual Result

no errors

### Additional Info

Easy to fix - #10630

**Comment by ronami** [comment]
Seems like a bug to me 👍
